src/vanilla/utils/proxyMap-indexMap.ts

```diff
@@ -0,0 +1,159 @@
+import { proxy, unstable_getInternalStates } from '../../vanilla.ts'
+const { proxyStateMap } = unstable_getInternalStates()
+const maybeProxify = (x: any) => (typeof x === 'object' ? proxy({ x }).x : x)
+const isProxy = (x: any) => proxyStateMap.has(x)
+
+type InternalProxyObject<K, V> = Map<K, V> & {
+  data: Array<[K, V | undefined]>
+  size: number
```

Comment:
just curious: do we need this?

Reply:
The type definition? I could declare it when I create `vObject` below if you prefer that

Reply:
I mean I thought `Map<K, V>` type already includes `size` property.